File: TransactionFrontend/TransactionFrontend/Models/TransactionDto.cs
using System.ComponentModel.DataAnnotations;

namespace TransactionFrontend.Models
{
    public class TransactionDto
    {
        public long Id { get; set; }
        [Range(0.01, double.MaxValue, ErrorMessage = "amount must be greater than zero.")]
        public decimal Amount { get; set; }
        public DateTime Timestamp { get; set; }
        [Required]
        public string TransactionType { get; set; }
        [Required]
        public string Status { get; set; }
        [Required]
        public string Currency { get; set; }
    }

    public class CreateTransactionDto
    {
        [Range(0.01, double.MaxValue, ErrorMessage = "amount must be greater than zero.")]
        public decimal Amount { get; set; }
        [Required]
        public string TransactionType { get; set; }
        [Required]
        public string Status { get; set; }
        [Required]
        public string Currency { get; set; }
    }
}

File: Transaction/src/main/java/frankfurt/transaction/Entity/TransactionDTO.java
package frankfurt.transaction.Entity;
import jakarta.validation.constraints.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
public class TransactionDTO {

    private Long id;

    @NotNull(message = "Amount is required")
    @DecimalMin(value = "0.01", message = "Amount must be greater than zero")
    private BigDecimal amount;

    private LocalDateTime timestamp;

    @NotNull(message = "Transaction type is required")
    private Transaction.TransactionType transactionType;

    @NotNull(message = "Status is required")
    private Transaction.Status status;

    @NotNull(message = "Currency is required")
    private Transaction.Currency currency;
}